sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4521,6 +4523,20 @@ def get_storage_mounts_metadata(
                 storage_metadata, sync_on_reconstruction=False)
         return storage_mounts
 
+    def _skypilot_predefined_envs(
+            self, handle: CloudVmRayResourceHandle) -> Dict[str, str]:
+        """Returns the SkyPilot predefined environment variables."""
+        return {
+            'SKYPILOT_CLUSTER_INFO': json.dumps({
+                'cluster_name': handle.cluster_name,
+                'num_nodes': handle.launched_nodes,
+                'cloud': str(handle.launched_resources.cloud),
+                'region': handle.launched_resources.region,
+                'zone': handle.launched_resources.zone,
+                'spot': int(handle.launched_resources.use_spot),
```

Comment:
Do users currently require num_nodes / spot? If not, how about keeping this minimal for now?

Reply:
Sounds good. Keeping it minimal should be fine